src/shared/ui/Button/styled.ts

```diff
@@ -10,20 +10,57 @@ export const StButton = styled.button<TButton>`
   justify-content: center;
   box-sizing: border-box;
   cursor: pointer;
-  color: ${theme.colorsText.primary};
-  background-color: ${theme.colors.btnFirst};
+  color: ${props => {
+    switch (props.$designType) {
+      case 'primary':
+        return theme.colorsText.secondary;
+      case 'secondary':
+        return theme.colorsText.alternate;
+      default:
+        return theme.colorsText.secondary;
+    }
+  }};
+  background-color: ${props => {
+    switch (props.$designType) {
+      case 'primary':
+        return theme.colors.bgBtnFirst;
+      case 'secondary':
+        return theme.colors.bgBtnSecond;
+      default:
+        return theme.colors.bgBtnFirst;
+    }
+  }};
   height: ${theme.sizesHeight.button};
   width: ${theme.sizesWidth.button};
   border-radius: ${theme.borderRadius.round};
-  border: none;
+  border: ${props => {
+    switch (props.$designType) {
+      case 'primary':
+        return 'none';
+      case 'secondary':
+        return `1px solid ${theme.colors.borderSecond}`;
+      default:
+        return 'none';
+    }
+  }};
 
+  /* TODO поправить кнопки в разных состояних, когда дизайнеры скорректируют */
   &:hover {
-    background: ${theme.colors.btnSecond};
-    box-shadow: inset 0 0 2px ${theme.colors.border};
+    background-color: ${props => {
+      switch (props.$designType) {
+        case 'primary':
+          return theme.colors.bgBtnFirst;
+        case 'secondary':
+          return theme.colors.bgBtnSecond;
+        default:
+          return theme.colors.bgBtnFirst;
+      }
```

Comment:
```suggestion
      switch (props.$designType) {
        case 'primary':
        default:
          return theme.colors.bgBtnFirst;
        case 'secondary':
          return theme.colors.bgBtnSecond;
      }
```
Можно одинаковые return вот так собрать в общие условия